controlplane/internal/dal/dal.go

```diff
@@ -215,25 +214,33 @@ func (d *DAL) GetLatestDeployment(ctx context.Context, module string) (*Deployme
 	return d.loadDeployment(ctx, deployment)
 }
 
-// RegisterRunner registers a new runner.
+// UpsertRunner registers or updates a new runner.
 //
-// It will return ErrConflict if a runner with the same endpoint already exists.
-func (d *DAL) RegisterRunner(
-	ctx context.Context,
-	key ulid.ULID,
-	language string,
-	endpoint *url.URL,
-	deployment types.Option[ulid.ULID],
-) error {
-	var deploymentSQLKey pgtype.UUID
-	if deploymentKey, ok := deployment.Get(); ok {
-		deploymentSQLKey.Valid = true
-		deploymentSQLKey.Bytes = deploymentKey
-	}
-	_, err := d.db.RegisterRunner(ctx, sqltypes.Key(key), language, endpoint.String())
+// ErrConflict will be returned if a runner with the same endpoint and a
+// different key already exists.
+func (d *DAL) UpsertRunner(ctx context.Context, runner Runner) error {
+	var pgDeploymentKey pgtype.UUID
+	if dkey, ok := runner.Deployment.Get(); ok {
+		pgDeploymentKey.Valid = true
+		pgDeploymentKey.Bytes = dkey
+	}
+	deploymentID, err := d.db.UpsertRunner(ctx, sql.UpsertRunnerParams{
+		Key:           sqltypes.Key(runner.Key),
+		Language:      runner.Language,
+		Endpoint:      runner.Endpoint,
+		State:         sql.RunnerState(runner.State),
+		DeploymentKey: pgDeploymentKey,
+	})
+	// sqltypes.Key(key), language, endpoint.String())
```

Comment:
And here. Maybe these are intentional?

Reply:
Aha whoops! These were leftover during refactoring!